src/split-criterions/GiniImpurity.jl

```diff
@@ -77,33 +75,33 @@ Args:
 Returns:
     A tuple containing the index of the best feature and the best split value.
 """
-function find_best_split(data::Matrix{T}, labels::Vector{L}, num_features_to_use::Int=-1) where {T, L}  
-    best_gini = Inf  
-    best_feature_index = -1  
-    best_feature_value = -1  
+function find_best_split(data::AbstractMatrix, labels::AbstractVector, num_features_to_use::Int=-1)
+    best_gini = Inf
+    best_feature_index = -1
```

Comment:
the default values for feature and threshold are 0 for information gain and -1 for gini impurity. We should be consistent and start with the same default values in our splitting functions

Reply:
see above